shub/settings.py

```diff
@@ -0,0 +1,537 @@
+    "SOCIAL_AUTH_LOGIN_REDIRECT_URL": "http://127.0.0.1",
+    # On any admin or plugin login redirect to standard social-auth entry point for agreement to terms
+    "LOGIN_REDIRECT_URL": "/login",
+    # DOMAIN NAMES
+    ## IMPORTANT: if/when you switch to https, you need to change "DOMAIN_NAME"
+    # to have https, otherwise some functionality will not work (e.g., GitHub webhooks)
+    "DOMAIN_NAME": "http://127.0.0.1",
+    "DOMAIN_NAME_HTTP": "http://127.0.0.1",
+    # Get additional admins from the environment
+    "HELP_CONTACT_EMAIL": "vsoch@users.noreply.github.com",
+    "HELP_INSTITUTION_SITE": "https://srcc.stanford.edu",
+    "REGISTRY_NAME": "Tacosaurus Computing Center",
+    "REGISTRY_URI": "taco",
+    # "UA-XXXXXXXXX"
+    "GOOGLE_ANALYTICS": None,
+    "GOOGLE_BUILD_SINGULARITY_VERSION": "v3.3.0-slim",
+    "SREGISTRY_DATABASE_ENGINE": "django.db.backends.postgresql_psycopg2",
+    "SREGISTRY_DATABASE_NAME": "postgres",
+    "SREGISTRY_DATABASE_USER": "postgres",
+    "SREGISTRY_DATABASE_HOST": "db",
+    "SREGISTRY_DATABASE_PORT": "5432",
+    # Minio Storage
+    "MINIO_SERVER": "minio:9000",  # Internal to sregistry
+    "MINIO_EXTERNAL_SERVER": "127.0.0.1:9000",  # minio server for Singularity to interact with
+    "MINIO_BUCKET": "sregistry",
+    "MINIO_REGION": "us-east-1",
+    # The rate limit for each view, django-ratelimit, "50 per day per ipaddress)
+    "VIEW_RATE_LIMIT": "50/1d",
+    "DJANGO_LOG_LEVEL": "WARNING",
+    # An image url or one of the following: 'mm', 'identicon', 'monsterid', 'wavatar', 'retro'. Defaults to 'mm'
+    "GRAVATAR_DEFAULT_IMAGE": "retro",
+}
+
+INTEGER_DEFAULTS = {
+    "API_PAGE_SIZE": 10,
+    # Set this to be some size in MB to limit uploads.
+    # Uploads > 2.5GB will not use memory, but the filesystem
+    "DATA_UPLOAD_MAX_MEMORY_SIZE": None,
+    # Limit users to N collections (None is unlimited)
+    "USER_COLLECTION_LIMIT": 2,
+    # The number of collections to show on the /<domain>/collections page
+    "COLLECTIONS_VIEW_PAGE_COUNT": 250,
+    # The maximum number of downloads allowed per container/collection, per week
+    "CONTAINER_WEEKLY_GET_LIMIT": 100,
+    "COLLECTION_WEEKLY_GET_LIMIT": 100,
+    "GOOGLE_BUILD_LIMIT": 100,
+    "GOOGLE_BUILD_TIMEOUT_SECONDS": None,  # None defaults to 10 minutes
+    "GOOGLE_BUILD_EXPIRE_SECONDS": 28800,
+    "CONTAINER_SIGNED_URL_EXPIRE_SECONDS": 10,
+    "MINIO_SIGNED_URL_EXPIRE_MINUTES": 5,
+}
+
+# Environment helpers
+
+
+def get_sregistry_envar(key):
+    envar_key = "SREGISTRY_%s" % key
+    return os.environ.get(envar_key)
+
+
+def get_sregistry_envar_list(key):
+    envar_list = os.environ.get(key) or []
+    if envar_list:
+        envar_list = [x.strip() for x in envar_list.split(",") if x.strip()]
+    return envar_list
+
+
+# Try reading in from secrets first
+try:
+    from .secrets import *
+
+# Fallback to dummy secrets (usually for testing)
```

Comment:
This would be surprising for me as a user and I would discourage us from importing an example file. It could mask a problem during development that would raise itself during production deployment.

Reply:
The only thing defined in the dummy secrets (after I tweak it) would be the `SECRET_KEY` - and this is mostly done to ensure that if someone clones the repository, does nothing, and brings it up with docker-compose there will be a secret key.  I think what I'm going to do is move the entirety of secrets to be alongside the main settings.py (since they are optional and can come from the environment) and then keep the dummy_secrets.py, but print a huge warning about the secret key not being modified. What do you think? Is there another or a better way to go about this?

Reply:
> and this is mostly done to ensure that if someone clones the repository, does nothing, and brings it up with docker-compose there will be a secret key

I would assert that it's better to hard fail with an exception if that is required and not set rather than using a testing value. That makes it both more secure and doesn't silently fail (or even worse: be used insecurely).

Reply:
How about:

```
# If we don't have a secret key, fire a warning
if "SECRET_KEY" not in locals():
    print("****** WARNING - using dummy secret key - you MUST change for production! ******")
    SECRET_KEY = "xxxxxxxxxxxxxxxxxx"
```
I understand the concern, but people's first impression of a piece of a software are hugely influenced by these interactions, and I think we can find a balance by putting enough verbosity alongside this (and ensuring there are ample docs, etc. that tell the person to set it).

Reply:
I get where you're coming from, but that makes it insecure by default and console warnings are way too easy to overlook.

Is it better to work out-of-the-box but risk a security vulnerability in production or start with a solid security posture and loudly fail with an error on how to fix it out-of-the-box?

And as a final data point, the current version doesn't work out-of-the-box without this set, so we don't "lose" anything by maintaining that approach.

Reply:
okay that's fair! I'll adjust to:

```python
# If we don't have a secret key, no go
if "SECRET_KEY" not in locals():
    sys.exit("SECRET_KEY is required but not set. Set SREGISTRY_SECRET_KEY.")
```